sdk/include/reactos/compat_undoc.h

```diff
@@ -0,0 +1,55 @@
+#ifndef COMPAT_UNDOC_H
+#define COMPAT_UNDOC_H
+
+#include "compatguid_undoc.h"
+
+
+typedef struct _ReactOS_ShimData
+{
+    DWORD dwReserved1[130];
+    DWORD dwSize;
+    DWORD dwMagic;
+    DWORD dwReserved2[243];
+    GUID CompatContext;
+} ReactOS_ShimData;
+
+
+#define REACTOS_SHIMDATA_MAGIC  0xAC0DEDAB
+
+#ifndef WINVER_VISTA
+#define WINVER_VISTA    0x0600
+#define WINVER_WIN7     0x0601
+#define WINVER_WIN8     0x0602
+#define WINVER_WIN81    0x0603
+#define WINVER_WIN10    0x0a00
+#endif
+
+static
+inline
+DWORD RosGetProcessCompatVersion(VOID)
+{
+    static DWORD g_CompatVersion = 0xffffffff;
+    if (g_CompatVersion == 0xffffffff)
+    {
+        ReactOS_ShimData* pShimData = (ReactOS_ShimData*)NtCurrentPeb()->pShimData;
+        if (pShimData && pShimData->dwMagic == 0xAC0DEDAB && pShimData->dwSize == sizeof(ReactOS_ShimData))
+        {
+            if (RtlCompareMemory(&pShimData->CompatContext, &COMPAT_GUID_VISTA, sizeof(GUID)) == sizeof(GUID))
```

Comment:
Would IsEqualGUID make sense here or does that introduce an unwanted dependency to something?

Reply:
Or RtlEqualLuid() ?

Reply:
A LUID is a 64 bit integer.

Reply:
That's an ole32 function.
I was looking for an Rtl version of it, couldnt find it and settled with {{RtlCompareMemory}}

Reply:
Yup that's fair. Exactly the kind of dependency I meant.